.github/workflows/cpp_package.yaml

```diff
@@ -175,5 +173,56 @@ jobs:
             - name: "Upload C++ packages"
               uses: actions/upload-artifact@v4
               with:
-                  name: cpp_mcu_bin-${{ runner.os }}-${{ runner.arch }}-${{ matrix.idf_target }}
+                  name: cpp_mcu_bin-${{ matrix.idf_target }}
                   path: ${{ runner.workspace }}/cppbuild/Slint-cpp-*
+
+    cmake_slint_compiler:
+        env:
+            CARGO_INCREMENTAL: false
+        strategy:
+            matrix:
+                os: [ubuntu-20.04, macOS-12, macos-14, windows-2022]
+                include:
+                    - os: ubuntu-20.04
+                      package_suffix: Linux-x86_64
+                    - os: macOS-12
+                      package_suffix: Darwin-x86_64
+                    - os: macos-14
+                      package_suffix: Darwin-arm64
+                    - os: windows-2022
+                      package_suffix: Windows
+
+        runs-on: ${{ matrix.os }}
+        steps:
+            - uses: actions/checkout@v4
+            - uses: ./.github/actions/install-linux-dependencies
+              with:
+                  old-ubuntu: true
+            - uses: ./.github/actions/setup-rust
+            # Build Slint-compiler with CMake so that CMakeLists.txt can configure C++ specific features
```

Comment:
i don't really follow. What C++ specific feature are you talking about? Why not do the same as for the viewer and lsp?

Reply:
Right now the build is the same as when doing `cargo build -p slint-compiler`, but I'd rather play safe for the day where say the CMake build activates a specific Cargo feature of tools/compiler/Cargo.toml.